src/common/transformations/src/transformations/common_optimizations/move_eltwise_up_data_movement.cpp

```diff
@@ -126,3 +130,93 @@ ov::pass::MoveEltwiseUpThroughDataMov::MoveEltwiseUpThroughDataMov(
     auto m = std::make_shared<ov::pass::pattern::Matcher>(eltwise_pattern, matcher_name);
     register_matcher(m, callback);
 }
+
+ov::pass::MoveEltwiseUpThroughDataMovPerChannel::MoveEltwiseUpThroughDataMovPerChannel() {
+    MATCHER_SCOPE(MoveEltwiseUpThroughDataMovPerChannel);
+
+    auto const_predicate = [](const ov::Output<ov::Node>& output) {
+        auto constant_op = std::dynamic_pointer_cast<ov::opset8::Constant>(output.get_node_shared_ptr());
+        if (!constant_op)
+            return false;
+
+        if (output.get_target_inputs().size() != 1)
+            return false;
+
+        const auto& shape = constant_op->get_shape();
+        return std::count_if(shape.begin(), shape.end(), [](size_t v) {
+                   return v > 1;
+               }) == 1;
+    };
+
+    auto eltw_predicate = [](const ov::Output<ov::Node>& output) {
+        if (output.get_target_inputs().size() != 1)
+            return false;
+
+        auto node = output.get_node();
+
+        if (node->get_input_partial_shape(0).size() < node->get_input_partial_shape(1).size())
```

Comment:
should we check that the ranks are not dynamic  here?

Reply:
for eltwise ops, we can't guarantee that we are checking `eltw_data_flow_in` vs `eltw_const_in` here
MatherPass automatically matches both options: BinaryElementwiseArithmetic(eltw_data_flow_in, eltw_const_in) and BinaryElementwiseArithmetic(eltw_const_in, eltw_data_flow_in)

https://github.com/openvinotoolkit/openvino/blob/b8ba9038d63cb933f81655326d23ba7a30421c4a/src/core/src/pattern/matcher.cpp#L153